Makefile.package-stanc

```diff
@@ -0,0 +1,116 @@
+OCAML_VER          =  $(shell opam info ocaml-base-compiler --field=installed-version --color=never)
+BASE_VER           =  $(shell opam info base --field=installed-version --color=never)
+BASE_BIGSTRING_VER =  $(shell opam info base_bigstring --field=installed-version --color=never)
+JANESTREET_VER     =  $(shell opam info jane-street-headers --field=installed-version --color=never)
+PPX_INLINE_VER     =  $(shell opam info ppx_inline_test --field=installed-version --color=never)
+BIN_PROT_VER       =  $(shell opam info bin_prot --field=installed-version --color=never)
+TIME_NOW_VER       =  $(shell opam info time_now --field=installed-version --color=never)
+PPX_EXPECT_VER     =  $(shell opam info ppx_expect --field=installed-version --color=never)
+CORE_KERNEL_VER    =  $(shell opam info ppx_expect --field=installed-version --color=never)
+
+RUNTIME_FILES = floats.c backtrace_byt.c alloc.c array.c bigarray.c \
+	backtrace.c str.c ints.c io.c compare.c dynlink.c stacks.c weak.c \
+	eventlog.c finalise.c gc_ctrl.c meta.c hash.c intern.c signals.c \
+	obj.c md5.c memprof.c memory.c extern.c parsing.c callback.c afl.c \
+	sys.c unix.c win32.c startup_aux.c startup_byt.c codefrag.c domain.c \
+	misc.c fix_code.c skiplist.c signals_byt.c minor_gc.c major_gc.c \
+	globroots.c compact.c lexing.c interp.c custom.c printexc.c debugger.c \
+	freelist.c roots_byt.c fail_byt.c
+
+CAML_FILES = threads.h unixsupport.h version.h
+
+BASE_FILES = hash_types/src/internalhash_stubs.c src/hash_stubs.c \
+	src/exn_stubs.c src/int_math_stubs.c
+
+JANESTREET_FILES = core_params.h ocaml_utils.h jane_common.h
+
+ifeq ($(DOWNLOAD_SOURCES),true)
+	OCAML_SRCDIR=ocaml_sources
+else
+	OCAML_SRCDIR=$(OCAML_LIBDIR)/../../.opam-switch/sources
+endif
+
+OCAML_LIBDIR=$(OPAM_SWITCH_PREFIX)/lib/ocaml
+COMPILER_SRCDIR=$(OCAML_SRCDIR)/ocaml-base-compiler.$(OCAML_VER)
+SEXLIB_DUNEPACKAGE=$(OCAML_LIBDIR)/../sexplib/dune-package
+
+stanc-bytecode:
+	# Temporarily remove the bigarray requirement from sexplib - otherwise unix is required
+	sed -i'.bak' -e 's/requires bigarray/requires/g' $(SEXLIB_DUNEPACKAGE)
+	# Temporarily mask the unix library so it can't be included
+	mv $(OCAML_LIBDIR)/unix.cma $(OCAML_LIBDIR)/unix.cma.bak
+
+	dune build src/stanc/stanc.bc.c
+	mv $(OCAML_LIBDIR)/unix.cma.bak $(OCAML_LIBDIR)/unix.cma
+	$(RM) $(SEXLIB_DUNEPACKAGE)
+	mv $(SEXLIB_DUNEPACKAGE).bak $(SEXLIB_DUNEPACKAGE)
```

Comment:
I believe if `dune build` fails this will leave the sexplib in a bad state.

There are probably ways to make this safer, but in general I don't really want anything in our makefiles to touch anything inside the opam folder

Reply:
(Note: if we did want to move this process to another repo, it would be reasonable IMO to use something like https://github.com/tarides/opam-monorepo and then edit the local version of the dependency. If it's going to be inside the `stanc3` repo, I don't want it to step on people's environments even by accident)